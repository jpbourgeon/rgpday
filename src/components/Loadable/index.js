import React from 'react'
import loadable from '@loadable/component'
import logger from 'src/logger'
import Loading from 'src/components/Loading'
import Default from 'src/pages/Default'

class ErrorBoundary extends React.Component {
  constructor (props) {
    super(props)
    this.state = { hasError: false }
  }

  static getDerivedStateFromError () {
    // Update state so the next render will show the fallback UI.
    return { hasError: true }
  }

  componentDidCatch (error, info) {
    // You can also log the error to an error reporting service
    if (error) logger.error('componentDidCatch', error)
    if (info) logger.info('componentDidCatch', info)
  }

  render () {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <Default />
    }
    return this.props.children
  }
}

const Loadable = (props) => {
  const { path, prefetch } = props
  let AsyncComponent
  if (prefetch) {
    AsyncComponent = loadable(() => import(/* webpackPrefetch: true */ `../../${path}`),
      { fallback: <Loading /> })
  } else {
    AsyncComponent = loadable(() => import(`../../${path}`),
      { fallback: <Loading /> })
  }
  return (
    <ErrorBoundary>
      <AsyncComponent />
    </ErrorBoundary>
  )
}

export default Loadable
