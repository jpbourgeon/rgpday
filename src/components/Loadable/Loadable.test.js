/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import Loadable from './'
import jssSerializer from 'src/utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The Loadable component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Loadable />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
