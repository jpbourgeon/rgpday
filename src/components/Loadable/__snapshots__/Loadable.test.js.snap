// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Loadable component should render correctly 1`] = `
<div
  className="container-3 container-1"
>
  <div
    className="item-4 12-43 row-2"
  >
    <div
      className="root-100 colorSecondary-104 indeterminate-102"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="svg-105"
        viewBox="22 22 44 44"
      >
        <circle
          className="circle-106 circleIndeterminate-108"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
